### PR TITLE
Separate regression system for splitting code

### PR DESCRIPTION
**Pull request by aehyvari**
This PR addresses issue #148 by introducing a separate regression testing system for the splitting code.

The tests have three possible outcomes: `unsat`, `sat`, and `unknown`, and the expected output is hardcoded in the test script.  The interesting case is when expected result is `unknown`: If the solver returns `unknown`, the output is tested by constructing a tree of literals corresponding to a split, i.e., a set of cubes.  The splits are tested for the following properties:
 - each cube must be consistent
 - there must not be duplicate literals in a cube
 - the tree must be binary
 - if a node has two children, one must be the positive and the other the negative literal
 - the tree must be balanced
 - the number of descendants of a node (not including the node) must be
    - 2^(D-d), where D is the depth of the tree and d is the depth of the node for an inner node
    - 0 for a leaf node.

The last condition is testing more the consistency of the script itself.

In addition an `unknown` result must produce a split, and non-`unknown` result must not produce a split

### CODE REVIEW

File: ci/run_travis_commands.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@ make install
 if [[ ${CMAKE_BUILD_TYPE} == Debug ]]; then
     cd ../regression && ./run-test-notiming.sh ../build/src/bin/opensmt;
     cd ../regression_itp && ./run-tests.sh ../build/src/bin/opensmt;
+    cd ../regression_splitting && ./bin/run_tests ../build/src/bin/opensmt;
```
Comment (blishko) [inline ci/run_travis_commands.sh:12]: Missing extension: `./bin/run_tests.sh`

Comment (aehyvari) [reply ci/run_travis_commands.sh]: Thanks!

File: regression_splitting/bin/check_result.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+def getLinesFromOutput(file):
+    lines = map(lambda x: x.strip(), open(file).readlines())
+    lines = filter(lambda x: re.search("; [-]?[0-9][0-9]*", x), lines)
+    lines = map(lambda x: x.split(), map(lambda x: x[2:], lines))
+    lines = [list(map(lambda x: int(x), l)) for l in lines]
+    for line in lines:
+        abses = set(map(abs, line))
+        if len(abses) != len(line):
+            die("inconsistent path")
+        if len(set(line)) != len(line):
```
Comment (aehyvari) [inline regression_splitting/bin/check_result.py:20]: This check seems to fail always if the above check fails, so is redundant.

File: regression_splitting/bin/check_result.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    if len(sys.argv) != 3:
+        print("Usage: %s <output-file> <expected>" % sys.argv[0])
+        sys.exit(1)
+
+    lines = getLinesFromOutput(sys.argv[1])
+    expected = sys.argv[2]
+
+    if len(lines) == 0 and expected == 'unknown':
+        die("Expected a split")
+        sys.exit(1)
```
Comment (aehyvari) [inline regression_splitting/bin/check_result.py:79]: Not reached.

File: regression_splitting/bin/check_result.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    if expected != 'unknown':
+        s = open(sys.argv[1]).read()
+        mo = re.search('^(sat|unsat)', s)
+        if not mo:
+            die("Expected %s but found no result" % expected)
+        if mo.group(1) != expected:
+            die("Expected %s but got %s" % (expected, mo.group(1)))
+        sys.exit(0)
+
```
Comment (aehyvari) [inline regression_splitting/bin/check_result.py:91]: Maybe add else and move sys.exit(0) to the end.

**Pull request merged**